scilpy/tractanalysis/fibertube_scoring.py

```diff
@@ -0,0 +1,541 @@
+        and the fiber in which it has been seeded.
+    """
+    mean_errors = []
+    error_tractogram = []
+
+    # objmode allows the execution of non numba-compatible code within a numba
+    # function
+    with objmode(centers='float64[:, :]', indices='int64[:, :]'):
+        centers, indices, _ = streamlines_to_segments(centerlines, False)
+    centers_fixed_length = len(centerlines[0])-1
+
+    # Building a tree for first fibertube
+    tree = nbKDTree(centers[:centerlines_length[0]-1])
+    tree_fi = 0
+
+    for si, streamline_fixed in enumerate(streamlines):
+        streamline = streamline_fixed[:streamlines_length[si]-1]
+        errors = []
+
+        seeded_fi = seeds_fiber[si]
+        fiber = centerlines[seeded_fi]
+        radius = diameters[seeded_fi] / 2
+
+        # Rebuild tree for current fiber.
+        if tree_fi != seeded_fi:
+            tree = nbKDTree(
+                centers[centers_fixed_length * seeded_fi:
+                        (centers_fixed_length * seeded_fi +
+                         centerlines_length[seeded_fi] - 1)])
+
+        # Querying nearest neighbor for each coordinate of the streamline.
+        neighbors = tree.query_parallel(streamline)[1]
+
+        for pi, point in enumerate(streamline):
+            nearest_index = neighbors[pi][0]
+
+            # Retrieving the closest cylinder segment.
+            _, pi = indices[nearest_index]
+            pt1 = fiber[pi]
+            pt2 = fiber[pi + 1]
+
+            # If we're within the fiber, error = 0
+            if (np.linalg.norm(point - pt1) < radius or
+                np.linalg.norm(point - pt2) < radius or
+                    point_in_cylinder(pt1, pt2, radius, point)):
+                errors.append(0.)
+            else:
+                distance, vector, segment_collision_point = dist_point_segment(
+                    pt1, pt2, point)
+                errors.append(distance - radius)
+
+                if return_error_tractogram:
+                    fiber_collision_point = segment_collision_point - (
+                        vector / np.linalg.norm(vector)) * radius
+                    error_tractogram.append([fiber_collision_point, point])
+
+        mean_errors.append(np.array(errors).mean())
+
+    return mean_errors, error_tractogram
+
+
+@njit
+def endpoint_connectivity(step_size, blur_radius, centerlines,
+                          centerlines_length, diameters, streamlines,
+                          seeds_fiber):
+    """
+    For every streamline, find whether or not it has reached the end segment
+    of its fiber.
+
+    VC: "Valid Connection": Contains streamlines that ended in the final
+        segment of the fiber in which they have been seeded.
+    IC: "Invalid Connection": Contains streamlines that ended in the final
+        segment of another fiber.
+    NC: "No Connection": Contains streamlines that have not ended in the final
+        segment of any fiber.
+
+    Parameters
+    ----------
+    step_size: any
```

Comment:
any ? float seems more appropriate  same for blur_radius and everywhere you used any :)